### ISSUE TITLE
Git-lfs?

### ISSUE THREAD

**Issue by Sergeeeek** [nodegit/nodegit#914]
Is it supported and if not, will it ever be?

**Comment by johnhaley81** [comment]
It is currently not supported but it will eventually.

**Comment by maxkorp** [comment]
Yeah, i know libgit2 supports clean/smudge filters, which is what lfs uses (was added by microsoft IIRC? specifically for lfs support for tfs-online i think).

**Comment by cedeon** [comment]
Not trying to come across as pushy but is there any progress on 'eventually' yet? :P <3

**Comment by zezba9000** [comment]
Iv'e figured out a bunch on how to use Git-LFS with libgit2Sharp.
I'm posting this here as I would like to see gitKraken get "kraken" on LFS features as we would use it for work if it did... otherwise we are sticking with Perforce unless my tool pans out.

Hope this helps in some way.

Git-LFS Clean/Smudge filter: https://github.com/zezba9000/Git-Game-GUI/blob/master/GitGameGUI/RepoUserControl.xaml.cs#L22

Get lfs blob file and not just its ptr for diff or merge: https://github.com/zezba9000/Git-Game-GUI/blob/master/GitGameGUI/Tools.cs#L196

Diff example: https://github.com/zezba9000/Git-Game-GUI/blob/master/GitGameGUI/ChangesUserControl.xaml.cs#L672

Merge example: https://github.com/zezba9000/Git-Game-GUI/blob/master/GitGameGUI/ChangesUserControl.xaml.cs#L486

**Comment by crusy** [comment]
+n

**Comment by raiscui** [comment]
+n

**Comment by StewMcc** [comment]
Any insight on whether this is on the horizon?

**Comment by dbelyaev** [comment]
+n

**Comment by chbecher** [comment]
+n

**Comment by zezba9000** [comment]
My guess is if people want this supported, asking the Git-LFS team to add a normal C API interface instead of just a CLI one will solve the problem.  The CLI interface was designed to work with Git-Core and nothing else. If you want it to run with libgit2, it needs to function as such. This is my personal experience with it.

**Comment by mojavelinux** [comment]
There appears to be a package that integrates nodegit with git-lfs. It uses a spawned child process, but for those looking for something, it might help:

https://yarnpkg.com/en/package/nodegit-lfs

**Comment by xmedeko** [comment]
Just to add to @mojavelinux post, node-git Github repo is https://github.com/Axosoft/nodegit-lfs
